src/java.desktop/share/native/libfontmanager/freetypeScaler.c

```diff
@@ -1285,10 +1285,18 @@ static int allocateSpaceForGP(GPData* gpdata, int npoints, int ncontours) {
     }
 
     /* failure if any of mallocs failed */
-    if (gpdata->pointTypes == NULL ||  gpdata->pointCoords == NULL)
+    if (gpdata->pointTypes == NULL || gpdata->pointCoords == NULL) {
+        if (gpdata->pointTypes != NULL)  {
+            free(gpdata->pointTypes);
+            gpdata->pointTypes = NULL;
+        }
+        if (gpdata->pointCoords != NULL) {
+            free(gpdata->pointCoords);
+            gpdata->pointCoords = NULL;
+        }
         return 0;
```

Comment:
If you wanted, you could streamline this to:
```
    if (gpdata->pointTypes == NULL || gpdata->pointCoords == NULL) {
        free(gpdata->pointTypes);
        free(gpdata->pointCoords);
        gpdata->pointTypes = gpdata->pointCoords = NULL;
        return 0;	
```

Since free(NULL) is valid and a noop.